conferences/ticketing_storage.go

```diff
@@ -0,0 +1,605 @@
+	claimPayments.Payment = processedPayments
+	return &claimPayments, nil
+}
+
+// updateClaimPayment saves the invoice and payments of this claim payment assuming it exists
+func updateClaimPayment(ctx context.Context, tx *sqldb.Tx, c *ClaimPayment) (*ClaimPayment, error) {
+	res, err := exec(ctx, tx,
+		`UPDATE claim_payment SET invoice = $1
+	WHERE id = $2`, c.Invoice)
+	if err != nil {
+		return nil, fmt.Errorf("updating payment for claims: %w", err)
+	}
+
+	ra, err := res.RowsAffected()
+	if err != nil {
+		return nil, fmt.Errorf("updating payment for claims: %w", err)
+	}
+	if ra == 0 {
+		return nil, fmt.Errorf("claim payment was not found")
+	}
+
+	processedPayments := make([]FinancialInstrument, len(c.Payment), len(c.Payment))
+
+	for i, cp := range c.Payment {
+		switch payment := cp.(type) {
+		case *PaymentMethodMoney:
+			processedPayments[i], err = insertMoneyPayment(ctx, tx, c.ID, payment)
+			if err != nil {
+				return nil, fmt.Errorf("inserting money payment: %w", err)
+			}
+		case *PaymentMethodConferenceDiscount:
+			processedPayments[i], err = insertDiscountPayment(ctx, tx, c.ID, payment)
+			if err != nil {
+				return nil, fmt.Errorf("inserting discount payment: %w", err)
+			}
+		case *PaymentMethodCreditNote:
+			processedPayments[i], err = insertCreditPayment(ctx, tx, c.ID, payment)
+			if err != nil {
+				return nil, fmt.Errorf("inserting credit payment: %w", err)
+			}
+		default:
+			return nil, fmt.Errorf("not sure how to process payments of type %T", cp)
+		}
+	}
+	newClaim := ClaimPayment{
+		ID:          c.ID,
+		ClaimsPayed: c.ClaimsPayed,
+		Payment:     processedPayments,
+		Invoice:     c.Invoice,
+	}
+	return &newClaim, nil
+}
+
+const attendeesToSlotTable = "attendees_to_slot_claims"
+
+// changeSlotClaimOwner changes the passed claims owner from source to target
+func changeSlotClaimOwner(ctx context.Context, tx *sqldb.Tx, slots []SlotClaim, source *Attendee, target *Attendee) (*Attendee, *Attendee, error) {
+	if source == nil || target == nil {
+		return nil, nil, fmt.Errorf("either source or target is undefined")
+	}
+
+	if len(slots) == 0 {
+		return nil, nil, fmt.Errorf("no slots to transfer")
+	}
+
+	if len(slots) > len(source.Claims) {
+		return nil, nil, fmt.Errorf("the passed source lacks those claims")
+	}
+
+	claimIDs := make([]interface{}, 0, len(slots))
+	claimIDsIndex := map[uint64]bool{}
+	idHolders := make([]string, 0, len(slots))
+	for i, slot := range slots {
+		if slot.ID == 0 {
+			return nil, nil, fmt.Errorf("some slot claims lack IDs, perhaps the have not been saved yet")
+		}
+		claimIDs = append(claimIDs, slot.ID)
+		claimIDsIndex[slot.ID] = true
+		idHolders = append(idHolders, fmt.Sprintf("%d", i+3))
+	}
+	args := append([]interface{}{target.ID, source.ID}, claimIDs...)
+	res, err := exec(ctx, tx,
+		fmt.Sprintf(`UPDATE attendees_to_slot_claims SET attendee_id = $1 WHERE attendee_id = $2 AND slot_claim_id IN (%s)`, strings.Join(idHolders, ",")),
```

Comment:
Can we use an extrapolated variable here instead? This might be vulnerable to SQL injection if a user was ever able to craft the ID.

Reply:
This inserts `$3, $4, .. $N` generated in the above loop, ids are passed as arguments.